contracts/Governance.sol

```diff
@@ -0,0 +1,943 @@
+        bytes32[] roles;
+        bytes32[] actions;
+        bytes32 descriptionHash;
+        Timers.BlockNumber voteStart;
+        Timers.BlockNumber voteEnd;
+        address[] targets;
+        address proposer;
+        bytes[] calldatas;
+        string[] signatures;
+        bool canceled;
+        bool executed;
+        mapping(address => Receipt) receipts;
+    }
+
+    /**
+     * @dev Receipt structure.
+     */
+    struct Receipt {
+        bool hasVoted;
+        uint8 support;
+        uint96[] votes;
+    }
+
+    IERC20Votes public token;
+    TimelockController private _timelock;
+
+    bytes32[] public rolesList;
+    mapping(uint256 => bytes32) private _timelockIds;
+    mapping(bytes32 => uint256) private _roles;
+    mapping(bytes32 => mapping(address => bool)) public votersRoles;
+    mapping(address => mapping(bytes32 => mapping(address => uint256)))
+        public delegations;
+    mapping(address => mapping(bytes32 => uint256)) public delegatees;
+    mapping(address => mapping(bytes32 => uint256)) public delegated;
+    mapping(address => mapping(bytes32 => bytes32)) public actionsRoles;
+    mapping(address => mapping(bytes32 => uint256)) public actionsQuorums;
+    mapping(bytes32 => uint256) public proposalThresholds;
+    mapping(bytes32 => address) public rolesChildrenDAO;
+
+    mapping(uint256 => Proposal) private _proposals;
+    mapping(address => uint256) private _nonces;
+
+    event ProposalCreated(
+        uint256 proposalId,
+        address proposer,
+        address[] targets,
+        uint256[] values,
+        bytes32[] roles,
+        bytes[] calldatas,
+        string[] signatures,
+        bytes32[] actions,
+        uint256 startBlock,
+        uint256 endBlock,
+        string description
+    );
+    event ProposalCanceled(uint256 proposalId);
+    event ProposalExecuted(uint256 proposalId);
+    event ProposalQueued(uint256 proposalId, uint256 eta);
+    event TimelockChange(address oldTimelock, address newTimelock);
+    event VoteCast(
+        address indexed voter,
+        uint256 proposalId,
+        uint8 support,
+        string reason
+    );
+
+    /**
+     * @dev Restrict access to governor executing address. Some module might override the _executor function to make
+     * sure this modifier is consistant with the execution model.
+     */
+    modifier onlyGovernance() {
+        require(_msgSender() == _executor(), 'Governor: onlyGovernance');
+        _;
+    }
+
+    modifier roleExists(bytes32 role) {
+        require(_roles[role] > 0, 'Governance::roleExists: role does not exist');
+        _;
+    }
+
+    modifier hasRole(bytes32 role, address delegatee) {
+        require(
+            votersRoles[role][delegatee],
+            'Governance::hasRole: delegatee does not have the role'
+        );
+        _;
+    }
+
+    constructor(IERC20Votes token_, TimelockController timelock_) EIP712(name(), version()) {
+        token = token_;
+        _timelock = timelock_;
```

Comment:
Doesn't `Governance` need to have the role of proposer in the `TimelockController` contract to be able to call `_timelock.scheduleBatch()` in `queue()`?
 
Where is that happening in Governance?

Reply:
You just set it in TimelockController on its deployment.